SW-4988: FE - Cohort Detail (view/edit) updates

joshbuchea opened the pull request:
This PR includes updates to the Cohort View/New/Edit views.

## Examples

### Before

#### View Cohort

![staging terraware io_accelerator_cohorts_1](https://github.com/terraware/terraware-web/assets/1474361/0b22837a-e7b7-4641-b720-d29551bf5abd)

#### Edit Cohort

![staging terraware io_accelerator_cohorts_1 (1)](https://github.com/terraware/terraware-web/assets/1474361/ca69f661-667b-41b6-9efd-a885f1bd1857)

### After

#### View Cohort

![localhost_accelerator_cohorts_1](https://github.com/terraware/terraware-web/assets/1474361/859e157c-9d0f-4cce-ad0f-39f5afa2de46)

#### Edit Cohort

![localhost_accelerator_cohorts_1_edit](https://github.com/terraware/terraware-web/assets/1474361/32e989a0-9f56-4471-a9f2-6737d29854e9)

karthikbtf commented on src/scenes/AcceleratorRouter/Cohorts/CohortForm.tsx:
```diff
@@ -102,29 +104,47 @@ export default function CohortForm<T extends CreateCohortRequestPayload | Update
-              required
-              selectedValue={localRecord.phase}
+
+          {/* TODO: uncomment this section once createdTime & modifiedTime are available in Cohort records */}
```
Create a ticket to follow up (here and the other TODOs in this PR)?

joshbuchea replied:
You got it 👍

karthikbtf reviewed (approved):
Looks good to me. Agree about extracting out those components for reuse.

joshbuchea commented:
Ok I removed the duplicate components and moved `ProjectField` into `src/components`. Dragging my feet on the rename though because I know I won't be happy with whatever I come up with 😅

karthikbtf commented:
> Ok I removed the duplicate components and moved `ProjectField` into `src/components`. Dragging my feet on the rename though because I know I won't be happy with whatever I come up with 😅

ProjectField sounds good. ProjectData? It's all good, we can always rename if we want to.

karthikbtf reviewed (approved):
Nice work!
